### PR TITLE
Add ansible-runner for stable-2.11

### PR DESCRIPTION
**Pull request by ashwini-mhatre**
Depends-On: https://github.com/ansible/ansible-runner/pull/650

### CODE REVIEW

File: .zuul.d/jobs.yaml [hunk 1]
```diff
@@ -133,6 +133,56 @@
...
+      - name: github.com/ansible/ansible
+        # NOTE (amhatre): Switch to stable-2.11 once branch is created.
+        override-checkout: devel
+    vars: &ansible_runner_image_vars_stable_2_11
+      container_images: &container_images_stable_2_11
+        - context: .
+          build_args:
+            - ANSIBLE_BRANCH=stable-2.11
+            # NOTE(pabelanger): There is no ansible-core 2.11, so we have to build ansible-base 2.11 ourself.
+            - ANSIBLE_CORE_IMAGE=quay.io/ansible/python-base:latest
```
Comment (pabelanger) [inline .zuul.d/jobs.yaml]: this can be dropped

File: .zuul.d/jobs.yaml [hunk 2]
```diff
@@ -133,6 +133,56 @@
...
 # =============================================================================
 
+- job:
+    name: ansible-runner-build-container-image-stable-2.11
+    parent: ansible-build-container-image
+    provides:
+      - ansible-runner-stable-2.11-container-image
+    requires:
+      - python-base-container-image
+      - python-builder-container-image
```
Comment (pabelanger) [inline .zuul.d/jobs.yaml]: this should be:

```
    requires:
      - ansible-core-container-image
```

File: .zuul.d/jobs.yaml (hunk 1 above)
Comment (ashwini-mhatre) [reply .zuul.d/jobs.yaml]: so we dont need  build_args key?
```suggestion
          build_args:
            - ANSIBLE_BRANCH=stable-2.11
            # NOTE(pabelanger): There is no ansible-core 2.11, so we have to build ansible-base 2.11 ourself.
            - ANSIBLE_CORE_IMAGE=quay.io/ansible/python-base:latest
```

Comment (pabelanger) [reply .zuul.d/jobs.yaml]: right, we can drop them in favor of the required ansible-core image

File: .zuul.d/jobs.yaml [hunk 3]
```diff
@@ -133,6 +133,51 @@
 
 # =============================================================================
 
+- job:
+    name: ansible-runner-build-container-image-stable-2.11
+    parent: ansible-build-container-image
+    provides:
+      - ansible-runner-stable-2.11-container-image
+    requires:
+      - ansible-core-container-image
```
Comment (pabelanger) [inline .zuul.d/jobs.yaml:142]: This really should be ansible-core-stable-2.11-container-image, which would come from:

https://github.com/ansible/ansible-core-image/blob/main/.zuul.d/jobs.yaml

However we haven't created a stable job before. If you want to try and hack on it great, I can likely build it maybe this evening

File: .zuul.d/jobs.yaml [hunk 4]
```diff
@@ -133,6 +133,51 @@
...
+    parent: ansible-runner-container-image-base
+
+- job:
+    name: ansible-runner-upload-container-image-stable-2.11
+    parent: ansible-upload-container-image
+    provides:
+      - ansible-runner-stable-2.11-container-image
+    requires:
+      - python-base-container-image
+      - python-builder-container-image
```
Comment (pabelanger) [inline .zuul.d/jobs.yaml]: this should be ansible-core image, as above

File: .zuul.d/jobs.yaml (hunk 3 above)
Comment (ashwini-mhatre) [reply .zuul.d/jobs.yaml:142]: i really not sure how to add that job. and what is the need for it? as in this job we are creating ansible-runner-stable-2.11-container-image. Right? So whats difference between ansible-runner-stable-2.11-container-image ,ansible-core-stable-2.11-container-image, ansible-core-container-image

**Comment by ashwini-mhatre** [comment]
recheck

File: .zuul.d/jobs.yaml (hunk 3 above)
Comment (pabelanger) [reply .zuul.d/jobs.yaml:142]: once we branch ansible/ansible for stable-2.11, the ansible-core-container-image will become 2.12 code base. So we'll need a 2nd new image.

That said, don't worry about it for now. I am going to refactor the job to remove the ansible-core image, as we don't use it downstream.

**Review by shanemcd — APPROVED** [review]

**Pull request merged**